idr/connections.py

```diff
@@ -15,55 +14,74 @@ def _configuration_from_url(config_url):
     OMERO binary protocol doesn't support load balancing nor session pinning
     so it has to be done client-side by connecting to a random server/port
     """
+    try:
+        host = re.match('\w+://([^:/]+)', config_url).group(1)
+    except AttributeError:
+        host = config_url
+        config_url = 'https://%s/connection/omero-client.json' % host
     r = requests.get(config_url)
     r.raise_for_status()
-    cfgs = r.json()
-    cfg = random.choice(cfgs)
-    return cfg
+    cfg = r.json()
+    return cfg, host
 
 
-def _lookup_parameter(initial, paramname, autocfg, default):
+def _lookup_parameter(initial, paramname, default):
     if initial is not None:
         return initial
-    v = autocfg.get(paramname)
-    if v is not None:
-        return v
     v = os.getenv('IDR_' + paramname.upper())
     if v is not None:
         return v
     return default
 
 
-def connection(host=None, user=None, password=None, port=None):
+def connection(host=None, user=None, password=None, port=None, verbose=1):
```

Comment:
Big :thumbsup: for this as opposed to me hack. I could imagine making `0` the default.